zipline/algorithm.py

```diff
@@ -445,6 +440,10 @@ def _create_generator(self, sim_params, source_filter=None):
                 data_portal=self.data_portal
             )
 
+        if not self.initialized:
```

Comment:
I can rewind the history a bit to find the exact algo, but IIRC it was because an algorithm needed access to a perf_tracker during initialize.

Note that between the lazy-mainline branch and master we made this change:

```
-        # Build a perf_tracker
-        self.perf_tracker = PerformanceTracker(sim_params=self.sim_params,
-                                               env=self.trading_environment)
-
+        self.perf_tracker = None
         # Pull in the environment's new AssetFinder for quick reference
```